### PR TITLE
Fix varies of style issues for community page

### PR DESCRIPTION
**Pull request by nekomeowww**
## Changes

1. Fixed overflowed line through of headings, use a better solution to achieve the same effect
2. Adjusted the font size of headings when in smaller screens, it was way too big for mobile devices and reduces readability
3. Added paddings for the Recent News section, it used to have no paddings and looks weird
4. Adjusted the navigation anchor links for sections, the button style was way too big and not harmony with the rest of the design pattern
5. Added comments for blocks
6. Make it more consistent for paddings, margins and other styles for each of the sections
7. Fixed the entangled styles with the size of view ports, make it more smoother when resizes
8. Remove hard-coded paddings and margins
9. Fixed incorrect vw unit usages, center navigation items, linted function calls
10. Added some small animation transitions for the smoother experience

## Pull Request Dependency

Requires

- [x] #43766 

#### Inconsistent Padding

image
image

#### Height of banner image is too small

image

#### Lines overflow

image

#### Inconsistent Padding

image

### Desktop

image
image

### Mobile

image
image
image
image

### CODE REVIEW

**Review by dipesh-rawat — COMMENTED** [review]
Based on the quick look of the [Preview Page](https://deploy-preview-43740--kubernetes-io-main-staging.netlify.app/community/), this looks like an improvement!
LGTM

[Preview Community Page](https://deploy-preview-43740--kubernetes-io-main-staging.netlify.app/community/)

**Review by Gauravpadam — COMMENTED** [review]
LGTM

A significant improvement! Thanks for this PR

**Comment by Gauravpadam** [comment]
@nekomeowww while we're at it, I can see some more improvements we can make (only if you wish to make them)

- The navigation items can be centred (with or without media queries)
- There's some persistent horizontal scrolling, If you can possibly target that?

Rest all good, Thanks again

**Comment by nekomeowww** [comment]
> @nekomeowww while we're at it, I can see some more improvements we can make (only if you wish to make them)
> 
> 
> 
> - The navigation items can be centred (with or without media queries)
> 
> - There's some persistent horizontal scrolling, If you can possibly target that?
> 
> 
> 
> Rest all good, Thanks again
> 
> 

Absolutely! Although, it's a bit late for tonight in my time zone, I will tweak it and get it squashed and updated tomorrow ASAP.

**Comment by nekomeowww** [comment]
> There's some persistent horizontal scrolling, If you can possibly target that?

Can you provide me some screenshot or steps to reproduce? On what size of the device are you experiencing  such horizontal scroll issue?

**Comment by Gauravpadam** [comment]
I'm on a viewport of ~1600px width

> Is this the navigation items you talked about?
Correct, In smaller viewports, Flex can act up and break the alignment as so
image

Perhaps some justify-content will help?

> 
image

**Comment by nekomeowww** [comment]
@Gauravpadam 

> Perhaps some justify-content will help?

Yes. Adjusted.

> 

Interesting. I've tested all sorts of the viewports all the way from 4K to 720P with or without refreshes, I never got the overflow scroll like you did.

Do you know how to use devtools to point the elements? May I ask you to help me to find the overflow element that causes the scroll?

**Comment by Gauravpadam** [comment]
> Do you know how to use devtools to point the elements? May I ask you to help me to find the overflow element that causes the scroll?

Sure, I'll try some tinkering and get back to you

File: static/css/community.css
Comment (Gauravpadam) [inline static/css/community.css:199]: Either this OR auto, As per your liking
```suggestion
  width: 100%;
```

File: static/css/community.css
Comment (Gauravpadam) [inline static/css/community.css:91]:
```suggestion
  width: 100%;
```

File: static/css/community.css
Comment (Gauravpadam) [inline static/css/community.css:172]:
```suggestion
  width: 100%;
```

File: static/css/community.css
Comment (Gauravpadam) [inline static/css/community.css:189]:
```suggestion
  width: 100%;
```

File: static/css/community.css
Comment (Gauravpadam) [inline static/css/community.css:266]: (OR auto)
```suggestion
  width: 100%;
```

File: static/css/community.css
Comment (Gauravpadam) [inline static/css/community.css:347]:
```suggestion
  width: 100%;
```

File: static/css/community.css
Comment (Gauravpadam) [inline static/css/community.css:153]: Let's stick with this
```suggestion
  max-width: 100%;
```

File: static/css/community.css
Comment (Gauravpadam) [inline static/css/community.css:103]: Can we add a `justify-content:center` after line 77?

**Review by Gauravpadam — COMMENTED** [review]
Addressing these should solve our problems

**Review by Gauravpadam — COMMENTED** [review]
Now that fits!

LGTM, Thanks for the good work @nekomeowww
(You'll have to wait a bit more for an approver's approval though)

**Comment by sftim** [comment]
On a wide viewport, these changes make the menu list harder to read (very big gaps between items).

## Before

Screenshot

---

## After

Screenshot



_however_, I don't object if the changes are overall a benefit. What do other reviewers think?

**Comment by sftim** [comment]
**Aside / doesn't block a merge**
The X (aka Twitter) content is a lot to scroll through; if we can shrink that and still have it useful, even better.

**Comment by nekomeowww** [comment]
> On a wide viewport, these changes make the menu list harder to read (very big gaps between items).

@sftim  I agree with you. I should keep the padding around navigation items to prevent it being to wide to read. I will adjust it as suggested and make it more closer to the previous design.

**Comment by nekomeowww** [comment]
> The X (aka Twitter) content is a lot to scroll through; if we can shrink that and still have it useful, even better.

@sftim I used the official support of height settings X (Twitter) to limit the height of the widget by setting the [`data-height` attribute](https://developer.twitter.com/en/docs/twitter-for-websites/timelines/overview#customization). However, such change requires updates of all the localizations of the community page, if limit the height by overriding `height` property with CSS property, then the border of the widget will be lost. Not sure if this solution is acceptable.

**Comment by sftim** [comment]
> I used the official support of height settings X (Twitter) to limit the height of the widget by setting the [`data-height` attribute](https://developer.twitter.com/en/docs/twitter-for-websites/timelines/overview#customization). However, such change requires updates of all the localizations of the community page, if limit the height by overriding `height` property with CSS property, then the border of the widget will be lost. Not sure if this solution is acceptable.

Let's do that, but in a [separate](https://github.com/kubernetes/website/pull/43740#issuecomment-1787336097) PR. If we have to change all localizations at once, make one PR that crosses localizations (add a note to the PR description to explain the exception)

**Comment by nekomeowww** [comment]
> Let's do that, but in a separate PR.

@sftim  Removed the changes that related to X (Twitter).

**Comment by nekomeowww** [comment]
image

Please note this "Nothing to see here - yet" empty list for Twitter (X) widget wasn't broke by either the related #43766 Pull Request and this Pull Request.

It is something the developer platform of Twitter (X) team should address and get fixed.

About the "Nothing to see here - yet" issue of Twitter (X) widget, there are two forums that are actively being used by community to report the incidents, read more and follow here:

- [Embedded Tweets Not Working - Publisher Tools / Twitter for Websites - Twitter Developers](https://twittercommunity.com/t/embedded-tweets-not-working/199477/56)
- [[Again] List Widget says “Nothing to see here - yet” if logged out - Publisher Tools / Twitter for Websites - Twitter Developers](https://twittercommunity.com/t/again-list-widget-says-nothing-to-see-here-yet-if-logged-out/198782/346)

**Comment by nekomeowww** [comment]
@sftim @Gauravpadam Could you please take a review on this?

**Review by aj11anuj — CHANGES_REQUESTED** [review]
Layout break problem is already fixed for main website of this page even for smaller viewports. But all the changes and enhancements of this PR seems betterment of whole page. Just one thing, if you can increase the size of headings a bit for smaller viewports, then it will look better.

Screenshot_2023-12-05-06-57-44-869_com android chrome

**Comment by nekomeowww** [comment]
> Layout break problem is already fixed for main website of this page even for smaller viewports. But all the changes and enhancements of this PR seems betterment of whole page. Just one thing, if you can increase the size of headings a bit for smaller viewports, then it will look better.

Sure. No problem.

But which one does everyone prefer?

### Use `24px`

image

### Use `28px`

image

**Pull request merged**
